Remotion/Data/DomainObjects/Mapping/IMappingConfiguration.cs

```diff
@@ -24,12 +24,11 @@ namespace Remotion.Data.DomainObjects.Mapping
   /// </summary>
   public interface IMappingConfiguration
   {
-    ClassDefinition[] GetTypeDefinitions ();
-    bool ContainsTypeDefinition (Type classType);
-    ClassDefinition GetTypeDefinition (Type classType);
-    ClassDefinition GetTypeDefinition (Type classType, Func<Type, Exception> missingTypeDefinitionExceptionFactory);
+    TypeDefinition[] GetTypeDefinitions ();
+    bool ContainsTypeDefinition (Type type);
+    TypeDefinition GetTypeDefinition (Type type, Func<Type, Exception> missingTypeDefinitionExceptionFactory);
+    bool ContainsClassDefinition (Type classType);
```

Comment:
Where's the usage for ContainsClassDefinition? (didn't do a checkout) It's assymetric to typedef so that's a question mark.

Reply:
The usages are in `IClassDefinitionProvider` (TypePipe infra) and in the `ObjectIDSerializer`. The methods are asymmetric as searching for a type definition using a class ID does not make much sense.

Reply:
Ah, I was only referring to `ContainsClassDefinition (Type classType)`. As far as I can tell, ObjectIDSerializer uses the classID-overload instead of the type.

Okay, In IClassDefinitionProvider we're struck because there's no good way to test for classdefinition via extension methods. Which means the interface has an asymetric api. should we approximate this via an extension method?
```
ContainsClassDefinition(type)
{
  if (!type.IsClass)
    return false;
  if (mapping.ContainsTypeDefinition(type))
    return mapping.GetTypeDefinition(type) is ClassDefinition);
  return false;
}
```

it's not perfect but it does make the api consistent. In theory, we could approximate that a class-type will always be part of classdef and we simplify the check a bit, but that's so...so...